catalyst_voices/lib/pages/account/account_popup.dart

```diff
@@ -0,0 +1,244 @@
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class AccountPopup extends StatelessWidget {
+  final String avatarLetter;
+  final VoidCallback? onProfileKeychainTap;
+  final VoidCallback? onLockAccountTap;
+
+  const AccountPopup({
+    super.key,
+    required this.avatarLetter,
+    this.onProfileKeychainTap,
+    this.onLockAccountTap,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return PopupMenuButton(
+      color: Theme.of(context).colors.elevationsOnSurfaceNeutralLv1White,
+      onSelected: (String? value) {
+        switch (value) {
+          case _profileAndKeychain:
+            onProfileKeychainTap?.call();
+            break;
+          case _lock:
+            onLockAccountTap?.call();
+            break;
+        }
+      },
+      itemBuilder: (BuildContext bc) {
+        return [
+          PopupMenuItem(
+            padding: EdgeInsets.zero,
+            enabled: false,
+            value: null,
+            child: _Header(
+              accountLetter: avatarLetter,
+              walletName: 'Wallet name',
+              walletBalance: '₳ 1,750,000',
+              accountType: 'Basis',
+              walletAddress: 'addr1_H4543...45GH',
+            ),
+          ),
+          const PopupMenuItem(
+            height: 48,
+            padding: EdgeInsets.zero,
+            enabled: false,
+            value: null,
+            child: _Section('My account'),
+          ),
+          PopupMenuItem(
+            padding: EdgeInsets.zero,
+            value: _profileAndKeychain,
+            child: _MenuItem(
+              'Profile & Keychain',
+              VoicesAssets.icons.userCircle,
+            ),
+          ),
+          PopupMenuItem(
+            padding: EdgeInsets.zero,
+            value: _lock,
+            child: _MenuItem(
+              'Lock account',
+              VoicesAssets.icons.lockClosed,
+              showDivider: false,
+            ),
+          ),
+        ];
+      },
+      offset: const Offset(0, kToolbarHeight),
+      child: IgnorePointer(
+        child: VoicesAvatar(
+          icon: Text(avatarLetter),
+        ),
+      ),
+    );
+  }
+}
+
+class _Header extends StatelessWidget {
+  const _Header({
+    required this.accountLetter,
+    required this.walletName,
+    required this.walletBalance,
+    required this.accountType,
+    required this.walletAddress,
+  });
+
+  final String accountLetter;
+  final String walletName;
+  final String walletBalance;
+  final String accountType;
+  final String walletAddress;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: _padding),
+          child: Row(
```

Comment:
Would i make sens to split each Column child into separate, private, widget?

for example `AccountTile` -> Row(avatar, wallet, accountType), `WalletAddressTile`

Reply:
If you are not strong about it, I would prefer to leave it as it is, because with extracting it would be too granular IMO